token/program/src/state.rs

```diff
@@ -286,3 +286,93 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
         _ => Err(ProgramError::InvalidAccountData),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_mint_unpack_from_slice() {
+        let src: [u8; 82] = [0; 82];
+        let mint = Mint::unpack_from_slice(&src).unwrap();
+        assert!(!mint.is_initialized);
+
+        let mut src: [u8; 82] = [0; 82];
+        src[45] = 2;
+        let mint = Mint::unpack_from_slice(&src).unwrap_err();
+        assert_eq!(
+            mint,
+            solana_program::program_error::ProgramError::InvalidAccountData
```

Comment:
nit: program error is already imported earlier, so these can be
```suggestion
           ProgramError::InvalidAccountData
```